### DISCUSSION TITLE
Running airflow in Docker - stuck on airflow-init

### DISCUSSION THREAD

**Discussion by ivelinhristovipsos** [apache/airflow#27123] [Q&A] [answered]
Hello,

I am following the https://airflow.apache.org/docs/apache-airflow/stable/howto/docker-compose/index.html guide to the letter, trying to run airflow in a dev environment. I am using a windows machine and docker is given 8gb of ram and 4 cores which should be plenty for only running airflow.

However, runing the
`docker-compose up airflow-init`
step, just results in the deployment getting stuck on this step forever:
```
[+] Running 4/5
 - Network airflow_default              Created                                                                                                                    0.1s
 - Volume "airflow_postgres-db-volume"  Created                                                                                                                    0.0s
 - Container airflow-redis-1            Created                                                                                                                    4.0s
 - Container airflow-postgres-1         Created                                                                                                                    4.0s
 - Container airflow-airflow-init-1     Creating                                                                                                                1140.4s
```
This last "Container airflow-airflow-init-1" just hangs and stays there forever. No errors, no logs, no nothing.
I've tried all the standard steps ->restarting/clearing the environment and pulling the compose file again, nothing works.

I'll appeciate ideas or suggestions?

**Comment by o-nikolas** [comment]
Hey Ivelin,

Have you added the `--verbose` flag to `docker-compose`? That might give you more logging that we can work with.

**Reply by ivelinhristovipsos** [reply]
Thank you for the suggestion, but the -`-verbose` flag desn't provide any more details in my case.
```
docker-compose --verbose up airflow-init
[+] Running 0/1
 - Container airflow-airflow-init-1  Creating                                                                    120.5s
```

**Reply by potiuk** [reply]
You need to look in your docker system logs (which I am not sure how to access on Winodows but if you are using Docker Desktop it should be easy in the UI I guess). 

Note that currently we have not tested (and we cannot attest) that Airflow in any form works on windows natively. The current "best" solution for Windows users is that they should use WSL2 and when they use Docker, the docker should be installed in WSL2 and the WSL2 docker shoudl be used. 

We are planning though to make it officially supported some time soon (the current status is tracked in https://github.com/apache/airflow/issues/10388 ) but the only way we can get it is to have people like you who try different things, find a solution and let us know, that can make us closer to make it "official".

**Comment by ivelinhristovipsos** [comment, marked as answer]
Thanks you all for your replies. After some additional testing, I was able to successfully run Airflow on Windows 10 and start fireing DAGs. 

Turns out that getting stuck on  `Container airflow-airflow-init-1  Creating ` had nothing to do with the containers or docker, it had to do with my computer setup - I had docker installed under my local admin account, in a repo that is also on my work OneDrive.
So when airflow-airflow-init-1 is executed there is a one off  user dialog asking for permissions for my local admin acc to access my regular users' OneDrive. This was not visible under my regular windows user, even when running docker ""as administrator". It was only visible when loggin into windows with my local admin and trying to fire the container.

TL;DR - It was not a problem with Docker or Airflow at all, it was a Windows/OneDrive user mistake.

**Reply by potiuk** [reply]
Good one. Adding to my toolbox of answers :).
